crates/egui_plot/src/plot_ui.rs

```diff
@@ -0,0 +1,328 @@
+        } else {
+            self
+        }
+    }
+
+    /// The pointer position in plot coordinates. Independent of whether the pointer is in the plot area.
+    pub fn pointer_coordinate(&self) -> Option<PlotPoint> {
+        // We need to subtract the drag delta to keep in sync with the frame-delayed screen transform:
+        let last_pos = self.ctx().input(|i| i.pointer.latest_pos())? - self.response.drag_delta();
+        let value = self.plot_from_screen(last_pos);
+        Some(value)
+    }
+
+    /// The pointer drag delta in plot coordinates.
+    pub fn pointer_coordinate_drag_delta(&self) -> Vec2 {
+        let delta = self.response.drag_delta();
+        let dp_dv = self.last_plot_transform.dpos_dvalue();
+        Vec2::new(delta.x / dp_dv[0] as f32, delta.y / dp_dv[1] as f32)
+    }
+
+    /// Read the transform between plot coordinates and screen coordinates.
+    pub fn transform(&self) -> &PlotTransform {
+        &self.last_plot_transform
+    }
+
+    /// Transform the plot coordinates to screen coordinates.
+    pub fn screen_from_plot(&self, position: PlotPoint) -> Pos2 {
+        self.last_plot_transform.position_from_point(&position)
+    }
+
+    /// Transform the screen coordinates to plot coordinates.
+    pub fn plot_from_screen(&self, position: Pos2) -> PlotPoint {
+        self.last_plot_transform.value_from_position(position)
+    }
+
+    /// Add a data line.
+    pub fn line(&mut self, mut line: Line) -> &mut Self {
+        if line.series.is_empty() {
+            return self;
+        };
+
+        // Give the stroke an automatic color if no color has been assigned.
+        if line.stroke.color == Color32::TRANSPARENT {
+            line.stroke.color = self.auto_color();
+        }
+        self.items.push(Box::new(line));
+        self
+    }
+
+    /// Add a polygon. The polygon has to be convex.
+    pub fn polygon(&mut self, mut polygon: Polygon) -> &mut Self {
+        if polygon.series.is_empty() {
+            return self;
+        };
+
+        // Give the stroke an automatic color if no color has been assigned.
+        if polygon.stroke.color == Color32::TRANSPARENT {
+            polygon.stroke.color = self.auto_color();
+        }
+        self.items.push(Box::new(polygon));
+        self
+    }
+
+    /// Add a text.
+    pub fn text(&mut self, text: Text) -> &mut Self {
+        if text.text.is_empty() {
+            return self;
+        };
+
+        self.items.push(Box::new(text));
+        self
+    }
+
+    /// Add data points.
+    pub fn owned_points(&mut self, mut points: Points<PlotPoints>) -> &mut Self {
+        // Give the points an automatic color if no color has been assigned.
+        if points.color == Color32::TRANSPARENT {
+            points.color = self.auto_color();
+        }
+        self.items.push(Box::new(points));
+
+        self
+    }
+
+    /// Add data points.
+    pub fn borrowed_points<'b, T: 'a>(&mut self, mut points: Points<T>) -> &mut Self
```

Comment:
It would be nice if we could combine `owned_points` and `borrowed_points`.

Reply:
I agree that it would be nicer.

We need to require that everything impls PlotItem and this is either done through a generic type constraint or well defined types that impl PlotItem.